### PR TITLE
[Path 18] Migrations and Seeders for users, donors and receivers

### LINKED ISSUE #18: Write migration, seeder for users, donors, receivers
implement migration, seeding for 
- users
- donors
- receivers

### PR DESCRIPTION
**Pull request by swornim00**
Fixes #18

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -15,9 +15,16 @@ public function up()
...
         Schema::create('users', function (Blueprint $table) {
             $table->increments('id');
-            $table->string('name');
+            $table->string('first_name');
+            $table->string('last_name');
             $table->string('email')->unique();
+            $table->string('username')->unique();
             $table->string('password');
+            $table->bigInteger('phone');
+            $table->longText('address');
```
Comment (samundra) [inline database/migrations/2014_10_12_000000_create_users_table.php:24]: use `string` with fixed length.

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 2]
```diff
@@ -15,9 +15,16 @@ public function up()
...
+            $table->string('gender');
+            $table->string('blood_group');
+            $table->integer('user_type_id');
```
Comment (samundra) [inline database/migrations/2014_10_12_000000_create_users_table.php]: add reference to `user_type` table. as in `$table->foreign('user_id')->references('id')->on('users');`

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 3]
```diff
@@ -15,9 +15,16 @@ public function up()
...
-            $table->string('name');
+            $table->string('first_name');
+            $table->string('last_name');
             $table->string('email')->unique();
+            $table->string('username')->unique();
             $table->string('password');
+            $table->bigInteger('phone');
+            $table->longText('address');
+            $table->string('gender');
+            $table->string('blood_group');
```
Comment (samundra) [inline database/migrations/2014_10_12_000000_create_users_table.php]: use `enum` we'll have fixed set of blood groups.

File: app/Donations.php [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Donations extends Model
+{
```
Comment (samundra) [inline app/Donations.php:9]: enable `soft deletes`

File: database/seeds/DonationsTableSeeder.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+          'receiver_id' => '4'
+        )
+      );
+      \App\Donations::create(
+        array(
+          'donor_id' => '2',
+          'receiver_id' => '4'
+        )
+      );
+      \App\Donations::create(
```
Comment (samundra) [inline database/seeds/DonationsTableSeeder.php:28]: - use faker to seed data
- don't use Fully qualified namespace. e.g. `\App\Donations`

```
use `App\Donations`;
```

File: database/seeds/UsersTableSeeder.php [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+     *
+     * @return void
+     */
+    public function run()
+    {
+
+      \App\User::create(array(
+        'first_name' => 'Swornim',
+        'last_name'=> 'Shrestha',
+        'email' => 'srestaswrnm@gmail.com',
```
Comment (samundra) [inline database/seeds/UsersTableSeeder.php]: use [Faker Library](https://github.com/fzaninotto/Faker) to generate random data. There are faker formatter for Nepali names as well.

File: database/seeds/UsersTableSeeder.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      ));
+
+
+      \App\User::create(array(
+        'first_name' => 'Pranisha',
+        'last_name'=> 'Shrestha',
+        'email' => 'stha_pro1@gmail.com',
+        'username' => 'pranisha',
+        'password' => \Hash::make('secret'),
+        'phone' => '9818278249',
```
Comment (samundra) [inline database/seeds/UsersTableSeeder.php]: Big NO NO for using the real data. Please replace these with faker.

**Review by samundra — CHANGES_REQUESTED** [review]
Make the changes that I have mentioned

**Comment by samundra** [comment]
Why `composer.lock` in commit ?

**Comment by swornim00** [comment]
Ah! Installed Faker!

**Comment by samundra** [comment]
@bravegurkha Please resolve the conflict

File: database/seeds/UsersTableSeeder.php [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        'gender' => 'Male',
+        'blood_group' => $blood_group[rand(0,7)],
+        'user_type_id' => '1',
+        'remember_token' => '',
+      ));
+
+
+      //User Data
+      foreach(range(1,15) as $x){
+
```
Comment (samundra) [inline database/seeds/UsersTableSeeder.php]: remove blank line

File: database/seeds/UsersTableSeeder.php [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      \App\User::create(array(
```
Comment (samundra) [inline database/seeds/UsersTableSeeder.php]: use 4 spaces for indentation. Please fix the indentation through the PR and use `phpcs` to verify that it meets our coding standard.

**Review by samundra — CHANGES_REQUESTED** [review]

File: app/UserDonation.php [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class UserDonation extends Model
+{
+    use SoftDelets;
```
Comment (samundra) [inline app/UserDonation.php]: typo mistake should be `SoftDeletes`

**Review by samundra — CHANGES_REQUESTED** [review]

**Comment by samundra** [comment]
Please fix the conflicts and address the comments

File: .gitignore [hunk 11]
```diff
@@ -5,4 +5,8 @@
 Homestead.json
 Homestead.yaml
 .env
+<<<<<<< HEAD
 .php_cs.cache
+=======
+composer.lock
+>>>>>>> master
```
Comment (samundra) [inline .gitignore]: these are due to merge conflict, we dont' need these. remove them.

**Review by samundra — APPROVED** [review]

**Pull request merged**